RFC @NixOS/golang, especially if you can get the test to run correctly. I'm not the biggest Gopher so I'm not sure how to handle the following error. Comaintainers are also very welcome. `passthru.updateScript` would also be good to add.

```
Running phase: checkPhase
--- FAIL: Test (0.31s)
    --- FAIL: Test/testdata/filterflag.txtar (0.01s)
        --- FAIL: Test/testdata/filterflag.txtar/L3 (0.01s)
            deadcode_test.go:117: deadcode failed: exit status 1 (stderr=deadcode: Load: err: exit status 1: stderr: go: -mod may only be set to readonly when in workspace mode, but it is set>
                        Remove the -mod flag to use the default readonly value,
                        or set GOWORK=off to disable workspace mode.
                )
FAIL
FAIL    golang.org/x/tools/cmd/deadcode 0.870s
FAIL
```

--- a/pkgs/development/tools/gotools/default.nix
+++ b/pkgs/development/tools/gotools/default.nix
@@ -2,14 +2,14 @@
 
 buildGoModule rec {
   pname = "gotools";
-  version = "0.7.0";
+  version = "0.16.1";
 
   # using GitHub instead of https://go.googlesource.com/tools because Gitiles UI is to basic to browse
   src = fetchFromGitHub {
     owner = "golang";
     repo = "tools";
     rev = "v${version}";
-    sha256 = "sha256-z5XJ7tflOfDBtv4rp7WEjnHsXIyjNw205PhazEvaYcw=";
+    hash = "sha256-qFDi+d+2OuI+mMBceZiN+kJ0gPcfgXXRDrDDwqKeDOM=";
   };
 
   postPatch = ''
@@ -20,7 +20,7 @@ buildGoModule rec {
     rm -r cmd/getgo
   '';
 
-  vendorHash = "sha256-fp0pb3EcGRDWlSpgel4pYRdsPJGk8/d57EjWJ+fzq7g=";
+  vendorHash = "sha256-oOBdh4mK3x9HbxD00EDKLjFgd/4NQRlQXrnCigGOwLg=";
 
   doCheck = false;
 

